feat: add configurable startup_timeout parameter for Docker daemon

analytically opened the pull request:
Add support for configuring the Docker daemon startup timeout via the source configuration. This allows users to increase the timeout from the default 120 seconds when running on slower systems or under resource constraints.

taylorsilva reviewed (changes requested):
One test is failing. Can reproduce by running `docker build -t docker-image-resource --target tests .`

```
41.02 Summarizing 1 Failure:
41.02   [FAIL] Out [It] times out retrying dockerd
41.02   github.com/concourse/docker-image-resource/tests/out_test.go:95
41.02
41.02 Ran 35 of 35 Specs in 40.860 seconds
41.02 FAIL! -- 34 Passed | 1 Failed | 0 Pending | 0 Skipped
41.02 --- FAIL: TestDockerImageResource (40.86s)
```

analytically committed:
feat: add configurable startup_timeout parameter for Docker daemon

Add support for configuring the Docker daemon startup timeout via
the source configuration. This allows users to increase the timeout
from the default 120 seconds when running on slower systems or under
resource constraints.

Signed-off-by: Mathias Bogaert <mathias.bogaert@gmail.com>
